snapcraft/tests/test_sources.py

```diff
@@ -325,3 +334,22 @@ def test_get_tar_source_from_uri(self):
             with self.subTest(key=source):
                 self.assertEqual(
                     snapcraft.sources._get_source_type_from_uri(source), 'tar')
+
+    @unittest.mock.patch('snapcraft.sources.Git.pull')
+    def test_get_git_source_from_uri(self, mock_pull):
+        sources = [
+            'git://github.com:ubuntu-core/snapcraft.git',
+            'git@github.com:ubuntu-core/snapcraft.git'
+        ]
+
+        for source in sources:
+            with self.subTest(key=source):
+
+                options = MockOptions(source=source,source_type='git')
```

Comment:
Here you will have a pep8 error because you are missing a space after the comma.